python/lsst/ts/salobj/base_script.py

```diff
@@ -581,6 +617,21 @@ def do_setCheckpoints(self, data):
             self.set_state(state=ScriptState.STOPPING)
             await self._exit()
 
+    def next_supplemented_group_id(self):
+        """Return the group ID supplemented with a new subgroup.
+
+        The subgroup is an increasing integer, starting with 1.
+
+        Raises
+        ------
+        RuntimeError
+            If there is no group ID.
+        """
+        if not self.group_id:
+            raise RuntimeError("No group ID")
+        self._sub_group_id += 1
+        return f"{self.group_id}-{self._sub_group_id}"
```

Comment:
@ktlim are you okay with this form or would you prefer the sub group to be prepended? It's not relevant for AP in general since standard visits and alternate standard visits will not be changing the group ID at all, but it might be relevant for when AP tries to check a group ID (unless it filters on survey name first). OCPS and butler don't care since they treat the group ID string as being opaque.

Reply:
I'm OK with this.  I think Prompt Processing will need to filter on survey name, and sub-group-ids will only be used, if at all, by code that knows that they're there.

Reply:
I personally prefer a suffix in order to have group IDs sort, at least approximately.

They will not sort if there are more than 10 unless we format the sub-group ID (an integer) with leading zeros. If you want that then please tell me how many digits you want. 6 digits would handle even the most pathalogical script. I fear 2 might not be enough. I suspect anywhere in 3-5 would be fine.

Reply:
@ktlim also said privately that not having leading zeros is fine.